sn_networking/src/record_store.rs

```diff
@@ -282,57 +287,35 @@ impl NodeRecordStore {
 
     /// Prune the records in the store to ensure that we free up space
     /// for the incoming record.
-    ///
-    /// An error is returned if we are full and the new record is not closer than
-    /// the furthest record
-    fn prune_storage_if_needed_for_record(&mut self, r: &Key) -> Result<()> {
+    fn prune_storage_if_needed_for_record(&mut self) {
         let num_records = self.records.len();
 
         // we're not full, so we don't need to prune
         if num_records < self.config.max_records {
-            return Ok(());
+            return;
         }
 
         // sort records by distance to our local key
-        let furthest = self
-            .records
-            .keys()
-            .max_by_key(|k| {
-                let kbucket_key = KBucketKey::from(k.to_vec());
-                self.local_key.distance(&kbucket_key)
-            })
-            .cloned();
-
-        // now check if the incoming record is closer than our furthest
-        // if it is, we can prune
-        if let Some(furthest_record) = furthest {
-            let furthest_record_key = KBucketKey::from(furthest_record.to_vec());
-            let incoming_record_key = KBucketKey::from(r.to_vec());
-
-            if incoming_record_key.distance(&self.local_key)
-                < furthest_record_key.distance(&self.local_key)
-            {
-                info!(
-                    "{:?} will be pruned to make space for new record: {:?}",
-                    PrettyPrintRecordKey::from(&furthest_record),
-                    PrettyPrintRecordKey::from(r)
-                );
-                // we should prune and make space
-                self.remove(&furthest_record);
-
-                // Warn if the furthest record was within our distance range
-                if let Some(distance_range) = self.distance_range {
-                    if furthest_record_key.distance(&self.local_key) < distance_range {
-                        warn!("Pruned record would also be within our distance range.");
-                    }
-                }
-            } else {
-                // we should not prune, but warn as we're at max capacity
-                warn!("Maximum number of records reached. Current num_records: {num_records}");
-            }
-        }
+        let mut sorted_records: Vec<_> = self.records.keys().cloned().collect();
```

Comment:
I wonder if instead of sorting on every write, we check if this new record is farther than a previous farthest? And just store that distance? 

That might be computationally less strenuous? And lets it be one in one out in a simpler fashion?

Reply:
the incoming one could be much closer than the stored ones, which means even we recorded the `farthest record`, after the `one to one swapping for the incoming one`, we still need to figure out which one is the real farthest among the left ones.